### ISSUE TITLE
End-points

### ISSUE THREAD

**Issue by amanshawar** [FACK1/ReWorks#15]
# Edited/Improved version in the comments of this issue.

## Front-End paths
- [ ] Splash page:  `/`
- [ ] Upload photo:  `/uploadPhoto`
- [ ] Details photo page:  `/:id/getDetailsPage`
- [ ] Get Details item after uploaded  `/item`

- [ ] Login page: ` /loginPage`
- [ ] SignUp page:  `/signUpPage`
- [ ] All items of user page:  `/:id/items-list`
- [ ] Feedback page:  `/feedbackPage`

## Back-End
- [ ] **POST**  `/addPhotoDetails`
      Add details of photo  to data

- [ ] **DELETE**  `/:id/deleteItem`
      Delete item

- [ ] **GET** `/:id/getDetails`
Details of uploaded photo

- [ ] **PUT**  `/:id/edit-item`
      Edit  details of selected item

- [ ] **GET**  `/:id/items`
      Get all items of user

- [ ] **POST**  `/feedback`
      Add feedback options

- [ ] **POST**  `/signup`
- [ ] **POST** `/login`
- [ ] **GET**  `/logout`

## Third-Party
- [ ] Amason S3: **POST**  `/addToAmazon`

**Comment by jema28** [comment]
For the start page, you don't need `/Home`, `/` works.

**Comment by jema28** [comment]
All items of user page : 'GET:/:id/Items' change 'Items' to lowercase 'items'.

**Comment by m4v15** [comment]
Immediate feedback:

- For paths/endpoints and things personally I think camelCase looks a bit weird (how many times have you seen that when browsing the web) and I would go with kebab-case instead
- generally I think add **a lot** more detail:
  - what data is sent back and forth - if it's JSON what (roughly) will be in the JSON
  - what happens at each end point? Spell it out very clearly:
> we will add XXX to the mongoDB and then use the mongoId to add YYY to the airtable and then will send back `{success:true}` to the front end
  - when do these back end endpoints get hit by the front end - maybe add a bit of context
- basically add a lot more detail so that someone reading this can come here and understand what a why you made these decisions

**Comment by orjwan98** [comment]
## Back-End

- [ ] **POST**  `/add-item`
When the user clicks `continue` it will check if the user is still logged in or not.
If they are: Send a `POST` request to  `/add-item` to save data in the database. (info being sent is the item’s info along with the `user_id` of the logged in user. (Make sure to check if record already exists or not to prevent duplicates).
If they’re not, display the log-in page. 

- [ ] **GET** `/:id/get-details`
Once you select an item from {ItemList} it will render you to {GetDetails}.
Call GET /:id/getDetails endpoint to get all the details related to the selected item.
Display the details!.
 
- [ ] **PUT**  `/:id/edit-item`
Post the item id and the new values for the edited inputs (details) through this endpoint to our server
Update  the related item record in Airtable by calling its API and set the new data

- [ ] **GET**  `/items`
Call GET /:id/items endpoint to get All items and their details.
List all the items on the ItemsList component.

- [ ] **POST**  `/feedback`
Add feedback  data to DATABASE if he is logged in if not it will store in JSON file
Render allItems component and show All items for the user.

- [ ] **POST**  `/signup`
Add the username and password to database to DATABASE
Render login component

- [ ] **POST** `/login`
Get the data of the username from DATABASE
Encrypt the password and compare it will the password from the form
Render allItems component and show All items for the user.

- [ ] **GET**  `/logout`
Clear cokiee
Render login component
 
## Third-Party
- [ ] Amason S3: **POST**  `/upload-photo`
When the user clicks on upload photo button:
A `POST` request (which has our image) is sent from the client to our server at `/upload-photo`. 
The image is sent from our server to the Amazon server. 
A response is getting back to `/upload-photo` which has the image’s link.
